pkg/filesystem/subvolume.go

```diff
@@ -131,6 +139,36 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	return subvolumeNames
 }
 
+// getk8sRefSnapshotHandle returns the snapshothandle for k8s ref of the volume snapshots
+func getK8sRefSnapshotHandle(ctx context.Context, clientsets *k8sutil.Clientsets) map[string]snapshotInfo {
+
+	Snap, err := snapclient.NewForConfig(clientsets.KubeConfig)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	snapList, err := Snap.VolumeSnapshotContents().List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching snapshots: %v\n", err))
+	}
+
+	snapshotHandles := make(map[string]snapshotInfo)
+	for _, snap := range snapList.Items {
+		if snap.Status.SnapshotHandle != nil {
+			// get the snaps id from snapshot handle
+			splitSnapshotHandle := strings.SplitAfterN(*snap.Status.SnapshotHandle, "-", 6)
+			if len(splitSnapshotHandle) < 6 {
+				return nil
+			}
+			snapshotHandleId := splitSnapshotHandle[len(splitSnapshotHandle)-1]
```

Comment:
does this also take care of https://github.com/ceph/ceph-csi/blob/d5849a4801fa9f3383cefc4f96e18fcad420dc80/examples/cephfs/snapshotclass.yaml#L19C5-L19C23

Reply:
yes, it just compares the snap-id ignore the prefix which matches with snapshothandle.